src/libPMacc/PMaccConfig.cmake

```diff
@@ -66,16 +66,24 @@ if(NOT "${CMAKE_CXX_FLAGS}" MATCHES "-std=c\\+\\+11")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 endif()
 
-set(CUDA_ARCH sm_20 CACHE STRING "Set GPU architecture")
-string(COMPARE EQUAL ${CUDA_ARCH} "sm_10" IS_CUDA_ARCH_UNSUPPORTED)
-string(COMPARE EQUAL ${CUDA_ARCH} "sm_11" IS_CUDA_ARCH_UNSUPPORTED)
-string(COMPARE EQUAL ${CUDA_ARCH} "sm_12" IS_CUDA_ARCH_UNSUPPORTED)
-string(COMPARE EQUAL ${CUDA_ARCH} "sm_13" IS_CUDA_ARCH_UNSUPPORTED)
+set(CUDA_ARCH "20" CACHE STRING "Set GPU architecture (semicolon separated list, e.g. '-DCUDA_ARCH=20;35;60')")
 
-if(IS_CUDA_ARCH_UNSUPPORTED)
-    message(FATAL_ERROR "Unsupported CUDA architecture ${CUDA_ARCH} specified. "
-                       "SM 2.0 or higher is required.")
-endif(IS_CUDA_ARCH_UNSUPPORTED)
+foreach(PMACC_CUDA_ARCH_ELEM ${CUDA_ARCH})
+    string(REGEX MATCH "^[0-9]+$" PMACC_IS_NUMBER ${CUDA_ARCH})
+    if(NOT PMACC_IS_NUMBER)
+        message(FATAL_ERROR "Defined compute architecture '${PMACC_CUDA_ARCH_ELEM}' in "
+                            "'${CUDA_ARCH}' is not an integral number, use e.g. '20' (for SM 2.0).")
+    endif()
+    unset(PMACC_IS_NUMBER)
+
+    # set flags to create device code for the given architecture
+    set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
+        "--generate-code arch=compute_${PMACC_CUDA_ARCH_ELEM},code=sm_${PMACC_CUDA_ARCH_ELEM}")
```

Comment:
actually I am not sure if you are still JIT forward compatible. By skipping the `compute_XY` part in `code` you do not embed a *PTX* binary load image and only a *assembled SASS*  binary load image for the specific SM (via `sm_XY`), which means there is no PTX code to generate missing SASS for other (newer) SMs.
```

This is because your `arch=compute_XY` does *not* trigger PTX generation!
```
Normally, this option alone does not trigger assembly of the generated PTX for a real architecture (that is the role of nvcc option --gpu-code, see below); rather, its purpose is to control preprocessing and compilation of the input to PTX. 
```
 	

Specify the name of the NVIDIA GPU to assemble and optimize PTX for.

nvcc embeds a compiled code image in the resulting executable for each specified code architecture, which is a true binary load image for each real architecture (such as sm_50), and PTX code for the virtual architecture (such as compute_50).

During runtime, such embedded PTX code is dynamically compiled by the CUDA runtime system if no binary load image is found for the current GPU. 
```
http://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#options-for-steering-gpu-code-generation